Documentation: Add extra terminology to the key concepts glossary (#4838)
ynput/OpenPype: website/docs/artist_concepts.md

### Subset

> A published output from an asset results in a subset

Reviewer: ```suggestion
A published output from an asset results in a subset.
```

The subset type is referred to as [family](#family), for example a rig, pointcache, look.
A single asset can have many subsets, even of a single family, named [variants](#variant).

> By default a subset is named as a combination of family + variant, sometimes prefixed with the task name (like workfile)

Reviewer: ```suggestion
By default a subset is named as a combination of family + variant, sometimes prefixed with the task name (like workfile).
```

### Family

Each published [subset][#subset] can have exactly one family assigned to it. Family determines the type of data that the subset holds. Family doesn't dictate the file type, but can enforce certain technical specifications. For example OpenPype default configuration expects `model` family to only contain geometry without any shaders or joints when it is published.

### Task

> A task defines a work area for an asset an artist can work in. For example asset *hero* can have tasks *modeling* and *rigging*. Tasks also have types. Multiple tasks of the same type may exist in an asset. A task with type FX could for example appear twice as *fx_fire* and *fx_cloth*.

Reviewer: ```suggestion
A task defines a work area for an asset where an artist can work in. For example asset *characterA* can have tasks named *modeling* and *rigging*. Tasks also have types. Multiple tasks of the same type may exist on an asset. A task with type `fx` could for example appear twice as *fx_fire* and *fx_cloth*.
```

"Hero" might be confused with hero version.

Process of exporting data from your work scene to versioned, immutable file that can be used by other artists in the studio.

#### (Publish) Instance

> A publish instance is a single entry which defines a publish output. Publish instances persist with the workfile. This way we can expect that a publish from a newer workfile will produce similar consistent versioned outputs.

Reviewer: ```suggestion
A publish instance is a single entry which defines a publish output. Publish instances persist within the workfile. This way we can expect that a publish from a newer workfile will produce similar consistent versioned outputs.
```
